grifts.go

```diff
@@ -44,33 +44,35 @@ func printMiddleware(a *App) {
 
 func printMiddlewareByRoute(a *App) {
 	mws := map[string]string{}
+	// TODO: middleware is 'per App' can it be a loop for Apps?
```

Comment:
I thought App is supposed to be a singleton? There is even an open issue that in some cases it is not when it should be: https://github.com/gobuffalo/buffalo/issues/1653

Now there can be multiple instances of app within the same binary? Isn't this a major breaking change?

Reply:
The `App` is not a singleton even though the name could be interpreted as such. Previously, (also still) `App.Group()` and `App.Resource()` return a new `App` which is a child of the root App. So as the result, there are many apps under the main App. The problem is that any instance of `App` still has the whole function of `App` so it is confusing and users can write a code like `App.Group().Stop()` which stops the root app.

This PR (#2226) did not change much and kept the structure the same as today since I didn't want to break any existing apps using v0, but it could be changed in the future version v1.

This PR is an interim solution for moving the app structure from `App` has `App`s to `App` has `Home`s. (I think the name `Home` is not super perfect and I thought `Group` could be the best, but we already have `App.Group` as a function name)

See #2222 and #2226 for more details and context.